Testing shows that it still defaults to 1 for BT samples and that trying to set it to a non-negative number errors out at Kconfig as desired:

```
sebo@mach:~/zephyr/samples/bluetooth/mesh$ cb
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.5.2", minimum required is "3.4") 
-- Selected BOARD nrf52840_pca10056
Zephyr version: 1.12.99
Parsing Kconfig tree in /home/sebo/zephyr/Kconfig
Using /home/sebo/zephyr/boards/arm/nrf52840_pca10056/nrf52840_pca10056_defconfig as base
Merging /home/sebo/zephyr/samples/bluetooth/mesh/prj.conf
warning: SYSTEM_WORKQUEUE_PRIORITY (defined at kernel/Kconfig:328) was assigned the value "0" but got the value "-1". Check its dependencies in the 'menuconfig' interface (see the Application Development Primer section of the manual), or in the Kconfig reference at http://docs.zephyrproject.org/reference/kconfig/CONFIG_SYSTEM_WORKQUEUE_PRIORITY.html (which is updated regularly from the master branch)
warning: BT_MESH_LPN_RECV_DELAY (defined at subsys/bluetooth/host/mesh/Kconfig:289) was assigned the value "40" but got the value "". Check its dependencies in the 'menuconfig' interface (see the Application Development Primer section of the manual), or in the Kconfig reference at http://docs.zephyrproject.org/reference/kconfig/CONFIG_BT_MESH_LPN_RECV_DELAY.html (which is updated regularly from the master branch)
warning: BT_MESH_LPN_SCAN_LATENCY (defined at subsys/bluetooth/host/mesh/Kconfig:322) was assigned the value "30" but got the value "". Check its dependencies in the 'menuconfig' interface (see the Application Development Primer section of the manual), or in the Kconfig reference at http://docs.zephyrproject.org/reference/kconfig/CONFIG_BT_MESH_LPN_SCAN_LATENCY.html (which is updated regularly from the master branch)
warning: BT_L2CAP_RX_MTU (defined at subsys/bluetooth/host/Kconfig:165) was assigned the value "69" but got the value "". Check its dependencies in the 'menuconfig' interface (see the Application Development Primer section of the manual), or in the Kconfig reference at http://docs.zephyrproject.org/reference/kconfig/CONFIG_BT_L2CAP_RX_MTU.html (which is updated regularly from the master branch)
warning: BT_CTLR_LE_ENC (defined at subsys/bluetooth/controller/Kconfig:197) was assigned the value "n" but got the value "y". Check its dependencies in the 'menuconfig' interface (see the Application Development Primer section of the manual), or in the Kconfig reference at http://docs.zephyrproject.org/reference/kconfig/CONFIG_BT_CTLR_LE_ENC.html (which is updated regularly from the master branch)
warning: user value 0 on the int symbol SYSTEM_WORKQUEUE_PRIORITY (defined at kernel/Kconfig:328) ignored due to being outside the active range ([-256, -1]) -- falling back on defaults
Error: Aborting due to non-whitelisted Kconfig warning 'warning: user value 0 on the int symbol SYSTEM_WORKQUEUE_PRIORITY (defined at kernel/Kconfig:328) ignored due to being outside the active range ([-256, -1]) -- falling back on defaults'.
Note: If this warning doesn't point to an actual problem, you can add it to the whitelist at the top of /home/sebo/zephyr/scripts/kconfig/kconfig.py.
CMake Error at /home/sebo/zephyr/cmake/kconfig.cmake:119 (message):
  command failed with return code: 1
Call Stack (most recent call first):
  /home/sebo/zephyr/cmake/app/boilerplate.cmake:241 (include)
  CMakeLists.txt:3 (include)


-- Configuring incomplete, errors occurred!
sebo@mach:~/zephyr/samples/bluetooth/mesh$ 
```

--- a/subsys/bluetooth/Kconfig
+++ b/subsys/bluetooth/Kconfig
@@ -35,6 +35,11 @@ config BT_CUSTOM
 
 endchoice
 
+# The Bluetooth subsystem requires the system workqueue to execute at
+# a cooperative priority.
+config SYSTEM_WORKQUEUE_PRIORITY
+	range -256 -1
+
 if BT_HCI
 
 config BT_HCI_RAW

--- a/subsys/bluetooth/common/dummy.c
+++ b/subsys/bluetooth/common/dummy.c
@@ -11,13 +11,6 @@
 
 #include <zephyr.h>
 
-/* The Bluetooth subsystem requires the system workqueue to execute at a
- * cooperative priority to function correctly. If this build assert triggers
- * verify your configuration to ensure that cooperative threads are enabled
- * and that the system workqueue priority is negative (cooperative).
- */
-BUILD_ASSERT(CONFIG_SYSTEM_WORKQUEUE_PRIORITY < 0);
-
 /* The Bluetooth subsystem requires the Tx thread to execute at higher priority
  * than the Rx thread as the Tx thread needs to process the acknowledgements
  * before new Rx data is processed. This is a necessity to correctly detect

